7_react/react1/week2/todo_list_app/src/components/todoList.jsx

```diff
@@ -0,0 +1,52 @@
+const TodoList = ({ todos, setTodos }) => {
+  const handleComplete = (todo) => {
+    setTodos(
+      todos.map((item) => {
+        if (item.id === todo.id) {
+          return { ...item, completed: !item.completed };
+        }
+        return item;
+      })
+    );
+  };
+  const handleDelete = ({ id }) => {
+    if (todos.length === 1) {
+      setTodos([]);
+      alert("No items left");
+    } else {
+      setTodos(todos.filter((todo) => todo.id !== id));
+    }
+  };
+
+  return (
+    <div>
+      {todos.map((todo) => (
```

Comment:
Here is where you would have to implement the logic of rendering the "No items left" message instead of the todos, when there are no todos left. I am sure you will figure it out easily. ✋